BucStop/wwwroot/js/pong.js

```diff
@@ -8,35 +39,136 @@ canvas.height = 600;
 
 // Define basic measurements for game objects
 const grid = 15;
-const paddleWidth = grid * 5;
-const paddleHeight = grid;
+const paddleWidth = grid * 5; // Width of the paddles
+const paddleHeight = grid;    // Height of the paddles
 
-// Slower speed for top paddle
-let topPaddleSpeed = 3.6; // Slower speed for top paddle
-let bottomPaddleSpeed = 9; // Faster speed for bottom paddle
+// Set initial speeds for paddles and ball
+let topPaddleSpeed = 3.6; // Speed of the AI paddle
+let bottomPaddleSpeed = 9; // Speed of the player's paddle
+let ballSpeed = 4; // Speed of the ball
 
-let ballSpeed = 4;
+// Initialize scores and game state variables
 let playerScore = 0;
 let computerScore = 0;
-let gameActive = false;
-let totalTime = 180; // 3 minutes in seconds
+let gameActive = false; // Tracks if the game is currently active
+let totalTime = 30; // Total game time in seconds
+let timerId; // For setting and clearing the interval
+
+// Leaderboard logic
+let leaderboard = JSON.parse(localStorage.getItem('leaderboard')) || [];
+const maxLeaderboardEntries = 10;
+
+// Updates the leaderboard with the new score
+function updateLeaderboard(score, initials) {
+    // Assume leaderboard is an array of { score, initials } objects
+    leaderboard.push({ score, initials });
+    leaderboard.sort((a, b) => b.score - a.score);
+    leaderboard = leaderboard.slice(0, maxLeaderboardEntries); // Keep only top entries
+    localStorage.setItem('leaderboard', JSON.stringify(leaderboard));
```

Comment:
Like above, change the text in quotes to pong-leaderboard or something like that to track it separately. It will still only be local to the machine I think but it will at least be tracked separately